app/src/main/java/app/revanced/manager/ui/screen/DashboardScreen.kt

```diff
@@ -113,6 +115,27 @@ fun DashboardScreen(
         )
     }
 
+    val showManagerUpdateDialog by vm.prefs.showManagerUpdateDialog.getAsState()
+
+    if (showManagerUpdateDialog) {
+        var showDialog by rememberSaveable { mutableStateOf(true) }
+
+        if (showDialog) {
+            vm.updatedManagerVersion?.let {
```

Comment:
I think this nesting is avoidable by reading `showManagerUpdateDialog` in the viewmodel and conditionally setting a variable.

Reply:
What about this?

```kt
    val showManagerUpdateDialog by vm.prefs.showManagerUpdateDialog.getAsState()
    var showDialog by rememberSaveable { mutableStateOf(showManagerUpdateDialog) }

    if (showDialog) {
        vm.updatedManagerVersion?.let {
            AvailableUpdateDialog(
                onDismiss = { showDialog = false },
                setShowManagerUpdateDialog = {
                    vm.viewModelScope.launch {
                        vm.prefs.showManagerUpdateDialog.update(it)
                    }
                },
                onConfirm = onUpdateClick,
                newVersion = it
            )
        }
    }